src/physlr-molecules.cc

```diff
@@ -314,6 +326,49 @@ biconnectedComponents(graph_t& subgraph, vertexToComponent_t& vertexToComponent)
 	}
 }
 
+
+template <class Graph, class vertexIter, class edgeSet>
+void
+make_subgraph(Graph& g, Graph& subgraph, edgeSet edge_set, vertexIter vBegin, vertexIter vEnd)
+{
+    // //   Make a vertex-induced subgraph of graph g, based on vertices from vBegin to vEnd
+
+    // Add vertices into the subgraph but set indexOriginal the index of it in the source graph.
+    for (auto& vIter = vBegin; vIter != vEnd ; ++vIter)
+    {
+        auto u = boost::add_vertex(subgraph);
+
+        subgraph[u].name = g[*vIter].name;
+        subgraph[u].weight = g[*vIter].weight;
+		subgraph[u].indexOriginal = g[*vIter].indexOriginal;
+    }
+
+    // Iterate over pairs of vertices:
+    // check whether there exist an edge between their corresponding vertices in the graph.
+    graph_t::vertex_iterator vIter1, vIter2, vend1, vend2;
+    for (boost::tie(vIter1, vend1) = vertices(subgraph); vIter1 != vend1; ++vIter1)
+    {
+        for (boost::tie(vIter2, vend2) = vertices(subgraph); vIter2 != vend2; ++vIter2)
+        {
+            if (vIter1 != vIter2)
+            {
+		        std::unordered_map<
```

Comment:
Might want to benchmark `tsl::robin_map` to see which is faster

Reply:
https://tessil.github.io/2016/08/29/benchmark-hopscotch-map.html
seem to be faster. I will benchmark for out usage too.

Reply:
we aleady have the robin hash library under vendor so you can remove those files